source/common/http/filter/ext_authz.h

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "envoy/config/filter/http/ext_authz/v2/ext_authz.pb.h"
+#include "envoy/ext_authz/ext_authz.h"
+#include "envoy/http/filter.h"
+#include "envoy/local_info/local_info.h"
+#include "envoy/runtime/runtime.h"
+#include "envoy/upstream/cluster_manager.h"
+
+#include "common/common/assert.h"
+#include "common/ext_authz/ext_authz_impl.h"
+#include "common/http/header_map_impl.h"
+
+namespace Envoy {
+namespace Http {
+namespace ExtAuthz {
+
+/**
+ * Type of requests the filter should apply to.
+ */
+enum class FilterRequestType { Internal, External, Both };
+
+/**
+ * Global configuration for the HTTP authorization (ext_authz) filter.
+ */
+class FilterConfig {
+public:
+  // TBD(saumoh) Take care of grpc service != envoy_grpc()
```

Comment:
TODO instead of TBD (to follow codebase convention)

Reply:
This isn't a limitation any more. Removed the comment